lib/datasource.js

```diff
@@ -365,36 +365,60 @@ DataSource.prototype.setup = function(name, settings) {
     }
   }
 
-  dataSource.connect = function(cb) {
-    var dataSource = this;
-    if (dataSource.connected || dataSource.connecting) {
-      process.nextTick(function() {
-        cb && cb();
-      });
+  dataSource.connect = function(callback) {
+    var self = this;
```

Comment:
AFAIK, `=>` functions are much more performant than `var self = this`. Since this code will run on Node.js 4.x and newer, we should be using arrow functions instead of `self` as much as we can.